src/app.js

```diff
@@ -0,0 +1,275 @@
+  let editList = [] // [clipStart, clipEnd, ... ]: clip means segment to keep
+  let silenceStartMsgs = []
+  let silenceEndMsgs = []
+  ffmpeg.setLogger(({ message }) => {
+    if (message.includes('silence_start')) {
+      silenceStartMsgs.push(message)
+    } else if (message.includes('silence_end')) {
+      silenceEndMsgs.push(message)
+    }
+  }) // start logging after this line
+  await ffmpeg.run(
+    '-i',
+    mediaInfo.fileName,
+    '-af',
+    'silencedetect=n=-27dB:d=1',
+    '-f',
+    'null',
+    '-'
+  )
+  let minClipLength = mediaInfo.frameRate / 2 // 0.5 s
+  editList.push(0) // start the first clip with frame 0
+  for (let i = 0; i < silenceStartMsgs.length; i++) {
+    let clipEnd = Math.round(
+      parseFloat(silenceStartMsgs[i].split('silence_start:')[1]) *
+        mediaInfo.frameRate
+    )
+    let clipStart = Math.round(
+      parseFloat(silenceEndMsgs[i].split('silence_end:')[1]) *
+        mediaInfo.frameRate
+    )
+    if (clipEnd - editList[editList.length - 1] < minClipLength) {
+      editList.pop() // clip is too short, pop clipStart and do not push end
+    } else {
+      editList.push(clipEnd)
+    }
+    editList.push(clipStart)
+  }
+  let finalClipEnd = mediaInfo.frameCount
+  if (finalClipEnd - editList[editList.length - 1] < minClipLength) {
+    editList.pop()
+  } else {
+    editList.push(finalClipEnd) // end the final clip with final frame
+  }
+  return editList
+}
+
+// Get the metadata of the audio/video file
+async function getMediaInfo (mediaFile) {
+  let mediaInfo = {
+    // default values
+    fileName: mediaFile.name,
+    hasVideo: false,
+    hasAudio: false,
+    width: 1920,
+    height: 1080,
+    frameRate: 24,
+    frameCount: 0 // duration in frames
+  }
+  let rawLogMsgs = []
+  ffmpeg.setLogger(({ message }) => {
+    if (message.startsWith('D', 2) || message.startsWith('S', 4)) {
+      rawLogMsgs.push(message)
+    }
+  })
+  await ffmpeg.run('-i', mediaFile.name)
+
+  for (let logMsg of rawLogMsgs) {
+    if (logMsg.includes('Video')) {
+      mediaInfo.hasVideo = true
+      mediaInfo.width = parseInt(
+        logMsg
+          .split('x')
+          .reverse()[1]
+          .split(',')
+          .reverse()[0]
+      )
+      mediaInfo.height = parseInt(logMsg.split('x').reverse()[0])
+      mediaInfo.frameRate = parseFloat(logMsg.split(',').reverse()[3])
+      break
+    }
+  }
+  for (let logMsg of rawLogMsgs) {
+    if (logMsg.includes('Audio')) {
+      mediaInfo.hasAudio = true
+      break
+    }
+  }
+  if (!Number.isInteger(mediaInfo.frameRate)) {
+    // get exact drop-frame frame rate e.g. 29.97 fps -> 30000/1001 fps
+    mediaInfo.frameRate = (Math.ceil(mediaInfo.frameRate) * 1000) / 1001
+  }
+  let hrs = parseInt(rawLogMsgs[0].split(':')[1]) // duration
+  let mins = parseInt(rawLogMsgs[0].split(':')[2])
+  let secs = parseFloat(rawLogMsgs[0].split(':')[3])
+  mediaInfo.frameCount = Math.floor(
+    ((hrs * 60 + mins) * 60 + secs) * mediaInfo.frameRate
```

Comment:
edge case: what if the video is more than 24 hours long?

Reply:
Not worried. Cameras can't shoot for that long and WASM only handles files up to 2GB